src/main/java/org/symphonyoss/symphony/messageml/markdown/MarkdownParser.java

```diff
@@ -244,7 +244,13 @@ private void visit(KeywordNode keyword) {
 
   private void visit(MentionNode mention) {
     try {
-      Mention node = new Mention(parent, ++index, mention.getUid(), dataProvider);
+      Mention node;
+      if (StringUtils.isEmpty(mention.getPrettyName())) {
```

Comment:
This code will never be hit. Here's how Markdown parsing works:

- first out MD parser looks for the JSON node _entities_ (_MarkdownParser.enrichMarkdown()_)
- if found, it injects special sections with metadata from the entity into the original message. In the case of mentions, an "enriched" message will look like this: "Hello USER_FOLLOW1!". This is very hacky, but necessary.
- then, the resulting "enriched" message is passed onto the CommonMark parser, which calls our "visit" method to construct an MD tree

In the case of mentions, the "USER_FOLLOW" special section will be recognised as a "MentionNode" and turned into a MML "Mention", which will try to resolve the user based on their userId.

In short, either modify the "enrichMarkdown" methods (danger! beware of dragons!) or leave the MD parser as is.